presto-parquet/src/main/java/com/facebook/presto/parquet/reader/LongTimestampMicrosColumnReader.java

```diff
@@ -36,12 +33,7 @@ protected void readValue(BlockBuilder blockBuilder, Type type)
         if (definitionLevel == columnDescriptor.getMaxDefinitionLevel()) {
             long utcMillis = MICROSECONDS.toMillis(valuesReader.readLong());
             // TODO: specialize the class at creation time
-            if (type instanceof TimestampWithTimeZoneType) {
-                type.writeLong(blockBuilder, packDateTimeWithZone(utcMillis, UTC_KEY));
-            }
-            else {
-                type.writeLong(blockBuilder, utcMillis);
-            }
+            type.writeLong(blockBuilder, utcMillis);
```

Comment:
The real long type data for timestamp with tz stored in parquet file is of presto's format? This may lead to many problems. As guided by @tdcmeehan, we should store the data in Iceberg format.

Reply:
We may need to modify the writer as well for this, but I need to verify. I noticed that we were improperly packing the date in the previous logic which led to incorrect values being read.

For example the following set of queries produced bad results:

```
create table t(t TIMESTAMP WITH TIME ZONE);

INSERT INTO t VALUES TIMESTAMP '1980-12-08 0:10:0 America/Los_Angeles';

presto:tpch> SELECT * FROM t;
               t
-------------------------------
 +46764-05-25 18:40:01.825 UTC
```

Reply:
> We may need to modify the writer as well for this.

Agree, the bad results in your example seems to be the problem caused by writing data with the incorrect format. So we need to customize a dedicated writer logic for timestamp with tz, which transform the long values encoded in presto's format to long values encoded in iceberg's format.

Reply:
I made some changes that I thought would fix this, added the `unpackMillisUtc` method to a new `TimestampWithTimezoneValueWriter` that I added to parquet, but we are losing all timezone information in the unpacking bit shift. This theoretically shouldn't be a problem since we  are supposed to be storing `millisUtc` in the first section of the bits, and the last 12 are reserved for whatever timezone we want to display to the user. However this is clearly not the case, since the filter operations are failing, the program expects the millis UTC to be transformed based on the timezone info, to get an _actual_ millis UTC. 

This is some pretty misleading variable naming and I think it leaves us with two options...

1. leave the system the same and build methods on top of it that apply the timezone part of the ts with tz to the millis part to get a correct instant (and change the variable names)
2. change the way ts with tz is read so at the time of unpacking we already have the millis UTC correct.

Sorry if this is redundant, I couldn't find documentation about how iceberg stores ts w tz under the hood.

Reply:
Referring to https://iceberg.apache.org/spec/#file-system-operations, Iceberg defines timestamp tz as follows:

```
Timestamp, microsecond precision, with timezone

Timestamp values with time zone represent a point in time: values are stored as UTC and do not retain a source time zone (2017-11-16 17:10:34 PST is stored/retrieved as 2017-11-17 01:10:34 UTC and these values are considered identical).
```

So it just contains UTC values in micros and do not retain the source time zone.

By the way, did the filtering operation fail because we forgot to do the same convert for timestamp with tz in `ExpressionConverter.getIcebergLiteralValue(...)`?

Reply:
Not sure about the filtering operation. I had to add `TimestampWithTimezoneType` to the cases in expression converter but so far I have just treated it like the other date time types that are included. I tried adding unpacking to the expression converter for ts w tz and now we have the regular tests passing, but failing with the batch reader enabled. Looking into this now